aip/0152.md

```diff
@@ -0,0 +1,84 @@
+---
+aip:
+  id: 152
+  state: reviewing
+  created: 2020-04-27
+permalink: /152
+redirect_from:
+  - /0152
+---
+
+# Jobs
+
+Occasionally, APIs may need to expose a task that takes significant time to
+complete, and where a transient [long-running operation][aip-151] is not
+appropriate. For example, a task could need to run repeatedly, or have separate
+permissions for configuring the task as opposed to running it.
+
+## Guidance
+
+An API **may** define a `Job` resource to represent a particular task with
+distinct setup, configuration, and execution:
+
+```proto
+message WritingJob {
+  // The resource name for the writing job.
+  string name = 1 [(google.api.resource) = {
+    type: "library.googleapis.com/WritingJob"
+    pattern: "publishers/{publisher}/writingJobs/{writing_job}"
+  }];
+
+  // Additional configuration...
+}
+```
+
+- The name of the resource **must** end with the word "Job".
+- The service **should** define all five of the standard methods, and use them
+  as the primary way to configure the job.
+
+### Run method
+
+The service **must** define a `Run` custom method that executes the job:
```

Comment:
Imagine a job which is scheduled to run either repeatedly, or once but not right now:

- Is that a suitable candidate for "this sort of job" (as described by the AIP)?
- If so, would there actually be a Run method, or just a Create?

It's not immediately clear to me - based on "a task could need to run repeatedly" whether cron-like APIs are part of this AIP, or whether that's out of scope.

Reply:
I think it does include cron-like jobs. And it seems like there *could* be scenarios where you may not want a "run this right now" method. I'd suggest this becomes **should**.

Reply:
Fine with **should** here.